plutus-scb/src/Cardano/Node/MockServer.hs

```diff
@@ -1,109 +1,210 @@
-import qualified Wallet.Emulator          as EM
+module Cardano.Node.MockServer(
+    MockServerConfig(..)
+    , defaultConfig
+    , main
+    ) where
+
+import           Cardano.Node.API               (API)
+import           Control.Concurrent             (forkIO, threadDelay)
+import           Control.Concurrent.MVar        (MVar, newMVar, putMVar, takeMVar)
+import           Control.Lens                   (view, over)
+import           Control.Monad                  (forever, void)
+import           Control.Monad.Freer            (Eff, Member)
+import           Control.Monad.Freer.State      (State)
+import qualified Control.Monad.Freer.State      as Eff
+import           Control.Monad.Freer.Writer     (Writer)
+import qualified Control.Monad.Freer.Writer     as Eff
+import           Control.Monad.IO.Class         (MonadIO, liftIO)
+import           Control.Monad.Logger           (MonadLogger, logDebugN, logInfoN, runStdoutLoggingT)
+import           Data.Foldable                  (traverse_)
+import qualified Data.Map                       as Map
+import           Data.Proxy                     (Proxy (Proxy))
+import           Data.Text.Prettyprint.Doc      (Pretty (pretty))
+import           Data.Time.Units                (Second, toMicroseconds)
+
+import qualified Network.Wai.Handler.Warp       as Warp
+import           Servant                        ((:<|>) ((:<|>)), Application, NoContent (NoContent), hoistServer,
+                                                 serve)
+
+import           Language.Plutus.Contract.Trace (InitialDistribution)
+import qualified Language.Plutus.Contract.Trace as Trace
+
+import           Ledger                         (Slot, Tx)
+import qualified Ledger
+import qualified Ledger.Blockchain              as Blockchain
+
+import           Cardano.Node.RandomTx
+import           Cardano.Node.SimpleLog
+
+import           Plutus.SCB.Arbitrary           ()
+import           Plutus.SCB.Utils               (tshow)
+
+import qualified Wallet.Emulator                as EM
+import           Wallet.Emulator.Chain          (ChainEffect, ChainEvent, ChainState)
+import qualified Wallet.Emulator.Chain          as Chain
+import qualified Wallet.Emulator.MultiAgent     as MultiAgent
+
+data BlockReaperConfig =
+    BlockReaperConfig
+        { brcInterval :: Second
+        , brcBlocksToKeep :: Int
+        }
+
+data MockServerConfig =
+    MockServerConfig
+        { mscPort                :: Int
+        , mscSlotLength          :: Second
+        -- ^ Duration of one slot
+        , mscRandomTxInterval    :: Maybe Second
+        -- ^ Time between two randomly generated transactions
+        , mscInitialDistribution :: InitialDistribution
+        -- ^ Initial distribution of funds to wallets
+        , mscBlockReaper         :: Maybe BlockReaperConfig
+        -- ^ When to discard old blocks
+        }
+
+defaultConfig :: MockServerConfig
+defaultConfig =
+    MockServerConfig
+        { mscPort = 8082
```

Comment:
This port number really needs to be in the `.yaml` file eventually. (Just a passing comment. No need to make the change for this PR.)